dspace-api/src/main/java/org/dspace/administer/CreateAdministrator.java

```diff
@@ -114,91 +144,124 @@ protected void negotiateAdministratorDetails()
-                firstName = firstName.trim();
-            }
+                email = console.readLine();
+                if (!StringUtils.isBlank(email)) {
+                    email = email.trim();
+                } else {
+                    System.out.println("Please provide an email address.");
+                    continue;
+                }
 
-            System.out.print("Last name: ");
-            System.out.flush();
+                System.out.print("First name: ");
+                System.out.flush();
 
-            lastName = console.readLine();
+                firstName = console.readLine();
 
-            if (lastName != null) {
-                lastName = lastName.trim();
-            }
+                if (firstName != null) {
+                    firstName = firstName.trim();
+                }
 
-            ConfigurationService cfg = DSpaceServicesFactory.getInstance().getConfigurationService();
-            if (cfg.hasProperty("webui.supported.locales")) {
-                System.out.println("Select one of the following languages: "
-                        + cfg.getProperty("webui.supported.locales"));
-                System.out.print("Language: ");
+                System.out.print("Last name: ");
                 System.out.flush();
 
-                language = console.readLine();
+                lastName = console.readLine();
 
-                if (language != null) {
-                    language = language.trim();
-                    language = I18nUtil.getSupportedLocale(new Locale(language)).getLanguage();
+                if (lastName != null) {
+                    lastName = lastName.trim();
                 }
-            }
-
-            System.out.println("Password will not display on screen.");
-            System.out.print("Password: ");
-            System.out.flush();
 
-            password1 = console.readPassword();
+                if (cfg.hasProperty("webui.supported.locales")) {
+                    System.out.println("Select one of the following languages: "
+                        + cfg.getProperty("webui.supported.locales"));
+                    System.out.print("Language: ");
+                    System.out.flush();
 
-            System.out.print("Again to confirm: ");
-            System.out.flush();
+                    language = console.readLine();
 
-            password2 = console.readPassword();
+                    if (language != null) {
+                        language = language.trim();
+                        language = I18nUtil.getSupportedLocale(new Locale(language)).getLanguage();
+                    }
+                }
 
-            //TODO real password validation
-            if (password1.length > 1 && Arrays.equals(password1, password2)) {
-                // password OK
-                System.out.print("Is the above data correct? (y or n): ");
-                System.out.flush();
+                password = getPassword(console);
```

Comment:
This becomes redundant if the password prompt is moved out of the loop.

Reply:
will have to look into this